Sources/Controllers/Map/Layers/OAContextMenuLayer.mm

```diff
@@ -400,6 +400,21 @@ - (OATargetPoint *) getTargetPointCpp:(const void *)obj
     
     if (found.count > 0)
     {
+        if (found.count > 1)
+        {
+            // Sometimes there appears two copy of one point. We decided to filter it right here (in UI).
+            // https://github.com/osmandapp/OsmAnd-Issues/issues/533
+
+            NSMutableDictionary<NSString *, OATargetPoint *> *filtredPoints = [NSMutableDictionary new];
```

Comment:
This seems very complicated. OAPOI implements ```equals``` so you can simply create a set of OAPOI and add target objects only if oaPOI is not present in the set.